core/src/main/java/org/projectnessie/cel/extension/StringsLib.java

```diff
@@ -0,0 +1,298 @@
+  private static final String LAST_INDEX_OF = "lastIndexOf";
+  private static final String LOWER_ASCII = "lowerAscii";
+  private static final String REPLACE = "replace";
+  private static final String SPLIT = "split";
+  private static final String SUBSTR = "substring";
+  private static final String TRIM_SPACE = "trim";
+  private static final String UPPER_ASCII = "upperAscii";
+
+  public static EnvOption strings() {
+    return Library.Lib(new StringsLib());
+  }
+
+  @Override
+  public List<EnvOption> getCompileOptions() {
+    List<EnvOption> list = new ArrayList<>();
+    EnvOption option =
+        EnvOption.declarations(
+            Decls.newFunction(
+                CHAR_AT,
+                Decls.newInstanceOverload(
+                    "string_char_at_int", Arrays.asList(Decls.String, Decls.Int), Decls.String)),
+            Decls.newFunction(
+                INDEX_OF,
+                Decls.newInstanceOverload(
+                    "string_index_of_string",
+                    Arrays.asList(Decls.String, Decls.String),
+                    Decls.Int)),
+            Decls.newFunction(
+                LAST_INDEX_OF,
+                Decls.newInstanceOverload(
+                    "string_last_index_of_string",
+                    Arrays.asList(Decls.String, Decls.String),
+                    Decls.Int)),
+            Decls.newFunction(
+                LOWER_ASCII,
+                Decls.newInstanceOverload(
+                    "string_lower_ascii", Arrays.asList(Decls.String), Decls.String)),
+            Decls.newFunction(
+                REPLACE,
+                Decls.newInstanceOverload(
+                    "string_replace_string_string",
+                    Arrays.asList(Decls.String, Decls.String, Decls.String),
+                    Decls.String)),
+            Decls.newFunction(
+                SPLIT,
+                Decls.newInstanceOverload(
+                    "string_split_string", Arrays.asList(Decls.String, Decls.String), Decls.Dyn)),
+            Decls.newFunction(
+                SUBSTR,
+                Decls.newInstanceOverload(
+                    "string_substring_int", Arrays.asList(Decls.String, Decls.Int), Decls.String)),
+            Decls.newFunction(
+                TRIM_SPACE,
+                Decls.newInstanceOverload(
+                    "string_trim", Arrays.asList(Decls.String), Decls.String)),
+            Decls.newFunction(
+                UPPER_ASCII,
+                Decls.newInstanceOverload(
+                    "string_upper_ascii", Arrays.asList(Decls.String), Decls.String)));
+    list.add(option);
+    return list;
+  }
+
+  @Override
+  public List<ProgramOption> getProgramOptions() {
+    List<ProgramOption> list = new ArrayList<>();
+    ProgramOption functions =
+        ProgramOption.functions(
+            Overload.binary(CHAR_AT, Gaurds.callInStrIntOutStr(StringsLib::charAt)),
+            Overload.binary(INDEX_OF, Gaurds.callInStrStrOutInt(StringsLib::indexOf)),
```

Comment:
Nit: these "simple" ones could be replaced with `String::indexOf`

Reply:
It will simpler to use static function as `BinaryOp` or `FunctionOp` etc. And the code style will be consistent to other functions. It could be acceptable to wrap `String.indexOf` to a static function.

Reply:
I'm okay with this.